security/WFCORE-5466_Remove_Legacy_Vault_Integration.adoc

```diff
@@ -0,0 +1,311 @@
+* Domains will need to handle configurations containing vault expressions for older host controllers.
+
+The vault elements in the WildFly CLI config are removed from the next version of the schema.  The
+presence of the element in ALL versions of configuration will result in a parsing error being
+reported.  The parsing is more abrubt in the CLI as this is only used for configuring SSL so will
+need immediate attention.
+
+All vault integration code is deleted from WildFly Core reducing the dependence on PicketBox.
+
+All vault test utilities are deleted and removed from the testsuite.
+
+The `vault` galleon layer will be removed from the available galleon layers.
+
+=== Nice-to-Have Requirements
+
+N/A
+
+=== Non-Requirements
+
+This RFE is not quite sufficient to remove dependencies on PicketBox from WildFly Core.
+
+The legacy security subsystem also supports it's own vault definition, this is outside the
+scope of the task and will be handled with the removal of the subsystem.
+
+The core management vault resources are not transformed so configuration does not need to be
+retained for managing older host controllers.
+
+The following items identified during the removal of Vault integration will be handled
+independently in their own follow up tasks:
+
+* https://issues.redhat.com/browse/WFLY-14919[WFLY-14919 - Credential store expression resolution not usable for deployment descriptors.]
+* https://issues.redhat.com/browse/WFCORE-5490[WFCORE-5490 - Elytron Expression Resolution too late to handle system properties.]
+
+== Implementation Plan
+
+This task will need to be developed in stages, logically the first step is to remove the
+integration from WildFly Core then move to WildFly, however this would break CI for any WildFly
+use of vault.
+
+* An early PR to WildFly to compensate for forthcoming changes.
+* Submit main PR to WildFly Core removing the vault integration.
+* A third PR to WildFly completing the removal including documentation updates.
+
+During this task follow up activities will be identified, these should not block the progress
+of the initial PRs being merged.
+
+Follow up tasks may include:
+
+* Re-purposing test cases or developing new test cases where Elytron coverage is required.
+* Additional enhancements for feature parity in relation to removed features.
+
+== Test Plan
+
+The following table identifies the tests in WildFly Core and WildFly affected by the removal.
+
+.Test Case Updates
+|===
+|Project |Test Case |Action
+
+|WildFly Core
+|org.jboss.as.server.services.security.RuntimeVaultReaderUnitTestCase
+|Removed
+
+|WildFly Core
+|org.jboss.as.server.RuntimeExpressionResolverUnitTestCase
+|Removed
+
+|WildFly Core
+|org.jboss.as.server.deployment.DeploymentAddHandlerTestCase
+|Tweaked
+
+|WildFly Core
+|org.jboss.as.server.test.InterfaceManagementUnitTestCase
+|Tweaked
+
+|WildFly Core
+|org.jboss.as.server.parsing.SystemPropertiesParsingTest
+|Reduced
```

Comment:
@darranl I think this should either be Removed or Tweaked (to use credential store exp). This test is for https://issues.redhat.com/browse/WFCORE-4736 and removing the expressions makes it pointless.